impeller/entity/contents/text_contents.cc

```diff
@@ -78,14 +75,43 @@ std::optional<Rect> TextContents::GetCoverage(const Entity& entity) const {
   return bounds->TransformBounds(entity.GetTransformation());
 }
 
-static bool CommonRender(const ContentContext& renderer,
-                         const Entity& entity,
-                         RenderPass& pass,
-                         const Color& color,
-                         const TextFrame& frame,
-                         Vector2 offset,
-                         const std::shared_ptr<GlyphAtlas>& atlas,
-                         Command& cmd) {
+void TextContents::PopulateGlyphAtlas(
+    const std::shared_ptr<LazyGlyphAtlas>& lazy_glyph_atlas,
+    Scalar scale) const {
+  lazy_glyph_atlas->AddTextFrame(frame_, scale);
+}
+
+bool TextContents::Render(const ContentContext& renderer,
+                          const Entity& entity,
+                          RenderPass& pass) const {
+  auto color = GetColor();
+  if (color.IsTransparent()) {
+    return true;
+  }
+
+  auto type = frame_.GetAtlasType();
+  auto scale = entity.DeriveTextScale();
```

Comment:
Rather than using the render-time entity to derive the correct scale of the text, why not record the derived scale (or even the full matrix) to `TextContents` when it's being built by Aiks? We know the CTM in that moment. Then, we just use that value for both the rendering lookup + `PopulateGlyphAtlas` without having to worry about fuzzy lookup problems.

Reply:
`TextContents` could technically get shared between entities given it's a `shared_ptr`, but in practice we don't share contents between different rendering operations today.

And if we did, we'd likely only do this for color sources, not special rendering operations that cross color source concerns with geometry concerns like with text.